Sources/PayPalNativeCheckout/PayPalDelegate.swift

```diff
@@ -1,25 +1,24 @@
 import Foundation
-#if canImport(PaymentsCore)
 import PaymentsCore
-#endif
+import PayPalCheckout
 
 /// PayPal delegate to handle events from PayPalClient
 public protocol PayPalDelegate: AnyObject {
 
     /// Notify that the PayPal flow finished with a successful result
     /// - Parameters:
-    ///   - client: the PayPalClient associated with delegate
     ///   - didFinishWithResult: the successful result from the flow
-    func paypal(_ payPalClient: PayPalClient, didFinishWithResult result: PayPalResult)
+    func paypal(didFinishWithResult approvalResult: PayPalCheckout.Approval)
 
     /// Notify that an error occurred in the PayPal flow
     /// - Parameters:
-    ///   - client: the PayPalClient associated with delegate
     ///   - didFinishWithError: the error returned by the PayPal flow
-    func paypal(_ payPalClient: PayPalClient, didFinishWithError error: CoreSDKError)
+    func paypal(didFinishWithError error: CoreSDKError)
 
     /// Notify that the PayPal flow has been cancelled
     /// - Parameters:
     ///   - client: the PayPalClient associated with delegate
-    func paypalDidCancel(_ payPalClient: PayPalClient)
+    func paypalDidCancel()
+
+    func paypalDidShippingAddressChange(shippingChange: ShippingChange, shippingChangeAction: ShippingChangeAction)
```

Comment:
in all of this methods i would add back the `payPalClient`. In the case of `shippingChange` the merchant will need to perform an action on the `payPalClient`, so we can return that as a parameter in every method on `PayPalDelegate`, so that the merchant doesnt have to mantian a reference to the `PayPalClient`
It is also a good practice for iOS that the first argument of a delegate method should be the "object delegated" (in this case, `PayPalClient`) 
Reference: https://www.swiftbysundell.com/articles/delegation-in-swift/

Reply:
@jcnoriega for second part, you mean send `ShippingAction` on every callback?